### PR TITLE
Allow for period as a prefix and yaml extension for config file

### LINKED ISSUE #4102: Support .yaml extension for our configuration file
Currently we only support `.yml` extension, which make some users (including myself) to be confused.

Probably, we need to just modifiy this file (https://github.com/rtfd/readthedocs-build/blob/bf2d1267c9a3dfa0b94588f23549ec527d5d5532/readthedocs_build/config/config.py#L17) to add the new names with the `.yaml` extension.

(take into account that `.yaml` is the official extension of the YAML file: http://yaml.org/faq.html)

We need:

- [ ] modify that file pointed here
- [ ] update the docs to reflect the new file names
- [ ] test this manually
- [ ] write some unit tests

Related: #4100

### PR DESCRIPTION
**Pull request by StefanoChiodino**
Fixes #4102 bringing it over from the build repo.

Ref. https://github.com/rtfd/readthedocs-build/pull/48

### CODE REVIEW

File: readthedocs/config/config.py [hunk 1]
```diff
@@ -966,17 +966,11 @@ def load(path, env_config):
...
-        files = '{}'.format(', '.join(map(repr, CONFIG_FILENAMES[:-1])))
-        if files:
-            files += ' or '
-        files += '{!r}'.format(CONFIG_FILENAMES[-1])
-        raise ConfigError(
-            'No files {} found'.format(files),
-            code=CONFIG_REQUIRED,
-        )
+        raise ConfigError('No configuration file found',
+                          code=CONFIG_REQUIRED)
```
Comment (stsewd) [inline readthedocs/config/config.py]: Our style guide is something like

```python3
raise ConfigError(
    'Text',
    code=CONFIG_REQUIRED
)
```

File: readthedocs/config/tests/test_config.py [hunk 2]
```diff
@@ -137,6 +145,14 @@ def test_load_unknow_version(tmpdir):
     assert excinfo.value.code == VERSION_INVALID
 
 
+def test_yaml_extension(tmpdir):
+    """ Make sure it's capable of loading the 'readthedocs' file with a 'yaml' extension. """
```
Comment (stsewd) [inline readthedocs/config/tests/test_config.py]: There are spaces in the sides of this docstring

File: readthedocs/config/tests/test_config.py [hunk 3]
```diff
@@ -790,6 +806,13 @@ def test_raise_config_not_supported():
     assert excinfo.value.code == CONFIG_NOT_SUPPORTED
 
 
+@pytest.mark.parametrize("correct_config_filename",
+                         [prefix + "readthedocs." + extension for prefix in {"", "."}
+                          for extension in {"yml", "yaml"}])
```
Comment (stsewd) [inline readthedocs/config/tests/test_config.py]: We use singlequotes

**Comment by stsewd** [comment]
Also, there is a test that needs to be changed https://travis-ci.org/rtfd/readthedocs.org/jobs/415587982#L970

**Comment by StefanoChiodino** [comment]
Sorry this took me a while! Hopefully this should pass now :)

File: readthedocs/config/tests/test_config.py [hunk 4]
```diff
@@ -1620,3 +1643,6 @@ def test_submodules_recursive_explict_default(self):
         assert build.submodules.include == []
         assert build.submodules.exclude == []
         assert build.submodules.recursive is False
+
+
+
```
Comment (stsewd) [inline readthedocs/config/tests/test_config.py]: I guess this extra lines shouldn't be here

**Comment by agjohnson** [comment]
Looks great! Thanks a ton for porting the PR over!

**Pull request merged**